mwdb/core/rate_limit.py

```diff
@@ -1,32 +1,60 @@
-import time
-
-import redis
-from flask import g
-from werkzeug.exceptions import TooManyRequests
+from flask import g, request
+from flask_limiter import Limiter
 
+from mwdb.core.capabilities import Capabilities
 from mwdb.core.config import app_config
 
 
-def rate_limit(key, duration, limit):
-    class TooManyRequestsWithRetryAfter(TooManyRequests):
-        def __init__(self, retry_after):
-            self.retry_after = retry_after
-            super().__init__(
-                f"You are too fast. Wait {retry_after} seconds before next request."
+def rate_limit_enabled():
+    if (
+        app_config.mwdb.enable_rate_limit
+        and g.auth_user is not None
+        and not g.auth_user.has_rights(Capabilities.unlimited_requests)
+    ):
+        return True
+    else:
+        return False
+
+
+limiter = Limiter(
+    key_func=lambda: f"{g.auth_user.login}-{request.method}"
+    if rate_limit_enabled()
+    else None,
```

Comment:
That `if` on `key_func` also doesn't look correct. Let's take a look at `Limiter.__init__`:
```python
        if not key_func:
            warnings.warn(
                "Use of the default `get_ipaddr` function is discouraged."
                " Please refer to"
                " https://flask-limiter.readthedocs.org/#rate-limit-domain"
                " for the recommended configuration",
                UserWarning,
            )
        if global_limits:
            self.__raise_global_limits_warning()

        self._key_func = key_func or get_ipaddr
```
If key_func is None, Limiter uses IP address as a key instead of disabling the rate limiter.

Reply:
key_func never be None, because it is instance of <class 'function'> so when rate_limit_enabled() returns False Limiter doesn't uses IP address - limiter doesn't calculate limits- I tested that. Of course if we explicity set key_func = None it really uses IP address, but we don't do it.

Reply:
After talk with @psrok1 conception was changed.
If user is logged we use login in key_func, but if user is not logged IP address will be used